### PR TITLE
All claims va treatment date validations mg

### PR DESCRIPTION
**Pull request by markgreenburg**
## Description
Adds validation to ensure that VA facility treatment dates start after the first service period date. Since we only have month and year for treatment dates but full dates for service period dates, any two dates with the same month and year should pass validation.

## Testing done
- Tested locally
- Unit tests

## Screenshots
screen shot 2018-12-17 at 2 43 30 pm

## Acceptance criteria
- [x] Validates that treatment start date is after earliest service period start date.
- [x] Content matches latest in ticket
- [x] Same month and year should validate (i.e., service period start date of `2000-01-01` and treatment start date of `2000-01-XX` should pass validation.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/validations.js
Comment (markgreenburg) [inline src/applications/disability-benefits/all-claims/validations.js]: I admit this might be a bit paranoid

**Review by peggygannon — APPROVED** [review]
@markgreenburg Looks good. thanks!

File: src/applications/disability-benefits/all-claims/pages/vaMedicalRecords.js
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/pages/vaMedicalRecords.js]: `.concat`'s fine, but you could alternatively `.push` this onto the array since there's only one.

File: src/applications/disability-benefits/all-claims/tests/pages/vaMedicalRecords.unit.spec.jsx
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/tests/pages/vaMedicalRecords.unit.spec.jsx]: Is this duplicated?

File: src/applications/disability-benefits/all-claims/validations.js
Comment (cvalarida) [reply src/applications/disability-benefits/all-claims/validations.js]: Alternatively, you could do `if (!_.get('servicePeriods.length', formData.serviceInformation, false)) { return; }`. Wouldn't check that it's an array, but that should be alright.

**Review by cvalarida — APPROVED** [review]
Looks good. Made a couple little enhancement suggestions if you feel like it, but it's fine as-is. Really the only thing I think you should take a look at before merging is that duplicate test (assuming I'm reading it right).

File: src/applications/disability-benefits/all-claims/pages/vaMedicalRecords.js
Comment (markgreenburg) [reply src/applications/disability-benefits/all-claims/pages/vaMedicalRecords.js]: So actually `.push()` returns the new length of the array, so I don't think this would work, unless I set it up somewhere else, which I think is messier than the `concat` approach.

File: src/applications/disability-benefits/all-claims/tests/pages/vaMedicalRecords.unit.spec.jsx
Comment (markgreenburg) [reply src/applications/disability-benefits/all-claims/tests/pages/vaMedicalRecords.unit.spec.jsx]: Not anymore it's not! 😛

File: src/applications/disability-benefits/all-claims/pages/vaMedicalRecords.js
Comment (cvalarida) [reply src/applications/disability-benefits/all-claims/pages/vaMedicalRecords.js]: Blech. Right. Forgot about that. Sounds like `.concat` is the way to go. :+1:

**Pull request merged**
